group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -726,6 +726,9 @@ private void removeGroup(
                     ClassicGroup classicGroup = (ClassicGroup) group;
                     metrics.onClassicGroupStateTransition(classicGroup.currentState(), null);
                     break;
+                default:
+                    log.warn("Unexpected group type detected while removing groups: " + group.type());
```

Comment:
nit: Could we say `Removed group {} with an unknown type {}`?

Reply:
Does it still remove the group?

Reply:
Looking at the first instruction in this method, the group is removed from `groups`.